main.go

```diff
@@ -66,14 +66,28 @@ func getSourceMap(source string, headers []string, insecureTLS bool) (m sourceMa
 	if isURL(source) {
 		// If it's a URL, get it.
 		req, err := http.NewRequest("GET", source, nil)
+		tr := &http.Transport{}
 
 		if insecureTLS {
-			tr := &http.Transport {
-				TLSClientConfig: &tls.Config{InsecureSkipVerify : true},
+			tr := &http.Transport{
```

Comment:
This is `:=` but should be `=`,  I suspect this is the cause of your logic issues.